runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -1225,7 +1225,10 @@ class INTERPRETER_CLASS
 					J9Class *destComponentClass = ((J9ArrayClass *)destClazz)->componentType;
 
 					if (J9_IS_J9CLASS_FLATTENED(srcClazz) && J9_IS_J9CLASS_FLATTENED(destClazz) && J9_ARE_NO_BITS_SET(srcComponentClass->classFlags, J9ClassHasReferences) && J9_ARE_NO_BITS_SET(destComponentClass->classFlags, J9ClassHasReferences)) {
-						if (srcClazz == destClazz) {
+						if (srcClazz == destClazz
```

Comment:
Is there anything preventing us from setting `J9ClassIsFlattened` on null restircted array only ? If `J9ClassIsFlattened` is not set on the nullable array, the code won't even reach here.

Reply:
I'm trying that out now on a different branch. The ddr value types tests are fine with removing the flag from nullable arrays but there are some failures in ValueTypeUnsafeTests I'm looking at. Are you waiting on this change for your work @a7ehuo ? Otherwise I can add those changes to this pr as well since its related.

Reply:
Nevermind, they are passing now.